aten/src/ATen/native/cuda/TensorFactories.cu

```diff
@@ -46,6 +46,7 @@ Tensor& eye_out_cuda(Tensor& result, int64_t n, int64_t m) {
 Tensor empty_cuda(IntArrayRef size, const TensorOptions& options) {
   AT_ASSERT(options.backend() == at::Backend::CUDA);
   AT_ASSERT(!options.is_variable());  // is_variable should have been 'unpacked'  // TODO: remove this when Variable and Tensor are merged
+  AT_ASSERTM(!options.pinned_memory(), "Only dense CPU tensors can be pinned");
```

Comment:
This needs to be a CHECK. Note that the asserts above are valid asserts, because our dispatch code is supposed to make sure those two conditions are impossible.